sorts/quick_sort.ts

```diff
@@ -10,29 +10,39 @@ export const partition = (
   left: number = 0,
   right: number = array.length - 1
 ) => {
-  const pivot = array[Math.floor((right + left) / 2)];
-  let i = left;
+  const pivotIndex = choosePivot(left,right);
```

Comment:
In the previous implementation of `partition` function, the return value was sometimes wrong, because it did not properly account for the position of the `pivot` element. For instance, if we would have called `partition([2, 3, 1])`, we would have gotten `1` as the result, instead of `0`.

My proposed fix is to first move the `pivot` element to the right of the array, by interchanging it with `array[right]`. Then, we mutate our subarray (that does not contain the last `pivot` element) such that all the elements to the left of a position `i` are smaller or equal to `pivot` element and all the elements to the right of the position `i` are greater or equal to the `pivot` element. Then, we just interchange the `pivot` element with the element on position `i`.